### PR TITLE
Package: klaus + dependencies

### PR DESCRIPTION
**Pull request by matthiasbeyer**
**This is WIP!**

This PR adds `klaus`, the git viewer that just works, plus its dependencies.

While the package is already building, I cannot start it by now, as there are runtime errors. So please do not merge yet!

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Can someone please have a look?

@jagajaga maybe you?

The point is: With `buildInputs` as dependency list, this results in a runtime error. With `propagatedBuildInputs` though, it works. I know that `propagatedBuildInputs` shouldn't be used here, am I right?

So... how to tackle this?

**Comment by jagajaga** [comment]
cc @domenkozar

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3410,6 +3433,27 @@ let
...
+  httpauth = buildPythonPackage rec {
+    version = "0.2";
+    name = "httpauth-${version}";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/h/httpauth/${name}.tar.gz";
+      md5 = "78d1835a80955e68e98a3ca5ab7f7dbd";
+    };
+
+    buildInputs = with self; [ ];
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: Remove if you don't need this.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -3386,6 +3386,29 @@ let
...
+      url = "https://pypi.python.org/packages/source/h/humanize/${name}.tar.gz";
+      md5 = "e8473d9dc1b220911cac2edd53b1d973";
+    };
+
+    buildInputs = with self; [ mock ];
+
+    doCheck = false;
+
+    meta = {
+      description = "python humanize utilities";
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: I don't like the discription but maybe it's ok :walking:

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6610,6 +6654,27 @@ let
...
+      sha256 = "0qcbv3shz530mn53pdc68fx38ylz72033xsrz77ffi0cks32az2w";
+    };
+
+    propagatedBuildInputs = with self;
+      [ humanize httpauth dulwich pygments flask ];
+
+    meta = {
+      description = "The first Git web viewer that Just Works";
+      homepage    = "https://github.com/jonashaag/klaus";
+      #license     = licnses.mit; # I'm not sure about the license
```
Comment (jagajaga) [inline pkgs/top-level/python-packages.nix]: Typo `licnses` -> `licenses`.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (matthiasbeyer) [inline pkgs/top-level/python-packages.nix]: That's the description on github for the package... I don't know what would fit better...

**Comment by matthiasbeyer** [comment]
I will squash asap you tell me that everything is fine! :-)

**Comment by jagajaga** [comment]
If you have successfully tested it then squash and I'll merge.

**Comment by matthiasbeyer** [comment]
@jagajaga here we go.

**Pull request merged**